add new master warehouse, use it for userobjects

rwcarlsen opened the pull request.

rwcarlsen committed:
Create a new singe-stop-shop warehouse and put userobjects in it

ref #10896, #11533, #6719, #6254, others?

dschwen commented on framework/include/base/MooseApp.h:
```diff
@@ -19,6 +19,7 @@
 #include "OutputWarehouse.h"
 #include "RestartableData.h"
 #include "ConsoleStreamInterface.h"
+#include "TheWarehouse.h"
```
Really? "TheWarehouse"?

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/211060/) on e1e3e8f wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11846/site)

This comment will be updated on new commits.

rwcarlsen committed:
handle pre-post aux more carefully. Finalize user objects in smaller, ordered groups

This fixes the one case (in rattlesnake) where someone wrote code where
an elemental user object depends on a side user object having been
finalized first.

permcody commented:
This is a significant change. Tagging @idaholab/moose-team for review.
